controllers/openstackcluster_controller.go

```diff
@@ -450,9 +453,27 @@ func (r *OpenStackClusterReconciler) SetupWithManager(ctx context.Context, mgr c
 		).
 		Watches(
 			&source.Kind{Type: &clusterv1.Cluster{}},
-			handler.EnqueueRequestsFromMapFunc(util.ClusterToInfrastructureMapFunc(infrav1.GroupVersion.WithKind("OpenStackCluster"))),
+			handler.EnqueueRequestsFromMapFunc(func(o client.Object) []reconcile.Request {
+				requests := clusterToInfraFn(o)
+				if requests == nil {
+					return nil
+				}
+
+				c := &infrav1.OpenStackCluster{}
+				if err := r.Client.Get(ctx, requests[0].NamespacedName, c); err != nil {
+					log.V(4).Error(err, "Failed to get OpenStack cluster")
+					return nil
+				}
+
+				if annotations.IsExternallyManaged(c) {
+					log.V(4).Info("OpenStackCluster is externally managed, skipping mapping.")
+					return nil
+				}
```

Comment:
Isn't the very same done already by setting the predicate in l 477? So this func may be obsolete?

Reply:
Hi @chrischdi, see https://kubernetes.slack.com/archives/C8TSNPY4T/p1620749340371300

No, this is getting a stream of cluster object events and then mapping them to openstackClusters.
The below predicates are only working on the base object (in this case the cluster, which does not have the annotation - the openstackCluster has the annotation)

Reply:
Ah good point :-) thank you fopr clarification.